src/dotnet-test-nunit/TestRunner.cs

```diff
@@ -236,6 +236,12 @@ void SetupLabelOutput(TestExecutionListener listener)
                  ? _options.DisplayTestLabels.ToUpperInvariant()
                  : "ON";
 
+            listener.TestStarted += (sender, args) =>
+            {
+                if (labels == "ALL")
```

Comment:
I think the simplest way around the duplication, which I introduced in nunit/nunit#1536 is to write the label "whether you need it or not" by calling a method that tracks the last test name label written and suppresses the output if it would be a duplicate. With the introduction of immediate output, there are more opportunities for the duplication.

I moved the "All" label from test started to test finished, but that may be a mistake. One of the uses of `labels=All` is to detect hanging tests. So perhaps using test started isn't a bad idea after all.